zebra-state/src/service.rs

```diff
@@ -718,98 +635,53 @@ impl StateService {
-                    result = self.validate_and_commit(child).map_err(CloneError::from);
-                    if result.is_ok() {
-                        // Update the metrics if semantic and contextual validation passes
-                        metrics::counter!("state.full_verifier.committed.block.count", 1);
-                        metrics::counter!("zcash.chain.verified.block.total", 1);
-                    }
+                    new_parents.push(child_hash);
                 }
-
-                let _ = rsp_tx.send(result.clone().map(|()| child_hash).map_err(BoxError::from));
-                new_parents.push((child_hash, result));
             }
-        }
-    }
-
-    /// Check that the prepared block is contextually valid for the configured
-    /// network, based on the committed finalized and non-finalized state.
-    ///
-    /// Note: some additional contextual validity checks are performed by the
-    /// non-finalized [`Chain`](non_finalized_state::Chain).
-    fn check_contextual_validity(
-        &mut self,
-        prepared: &PreparedBlock,
-    ) -> Result<(), ValidateContextError> {
-        let relevant_chain = self.any_ancestor_blocks(prepared.block.header.previous_block_hash);
-
-        // Security: check proof of work before any other checks
-        check::block_is_valid_for_recent_chain(
-            prepared,
-            self.network,
-            self.disk.db.finalized_tip_height(),
-            relevant_chain,
-        )?;
-
-        check::nullifier::no_duplicates_in_finalized_chain(prepared, &self.disk.db)?;
 
-        Ok(())
+            let _ = non_finalized_block_write_sender
+                .send(NonFinalizedWriteCmd::FinishProcessingQueued);
+        };
     }
 
     /// Return the tip of the current best chain.
     pub fn best_tip(&self) -> Option<(block::Height, block::Hash)> {
         self.mem.best_tip().or_else(|| self.read_service.db.tip())
     }
 
-    /// Return the height for the block at `hash` in any chain.
-    pub fn any_height_by_hash(&self, hash: block::Hash) -> Option<block::Height> {
-        self.mem
-            .any_height_by_hash(hash)
-            .or_else(|| self.read_service.db.height(hash))
-    }
-
-    /// Return an iterator over the relevant chain of the block identified by
-    /// `hash`, in order from the largest height to the genesis block.
-    ///
-    /// The block identified by `hash` is included in the chain of blocks yielded
-    /// by the iterator. `hash` can come from any chain.
-    pub fn any_ancestor_blocks(&self, hash: block::Hash) -> block_iter::Iter<'_> {
-        block_iter::Iter {
-            service: self,
-            state: block_iter::IterState::NonFinalized(hash),
-        }
-    }
-
     /// Assert some assumptions about the prepared `block` before it is validated.
     fn assert_block_can_be_validated(&self, block: &PreparedBlock) {
         // required by validate_and_commit, moved here to make testing easier
```

Comment:
This comment needs to be updated, but maybe we want to move this method to the `write` module instead, so it is closer to the code it is related to?

```
        // required by validate_and_commit_non_finalized, moved here to make testing easier
```

Reply:
It looks like this is being used from the CommitBlock call before the block is queued. I think it makes more sense to update the comment to `/// Assert some assumptions about the prepared `block` before it is queued.` and leave it as a method on StateService.

Reply:
Updated the comment in [`a27aa3f`](https://github.com/ZcashFoundation/zebra/pull/5257/commits/a27aa3fa449dd9d9bd482f0de627f349c738d3b7), left it as a method in StateService since it's currently being used in the call fn and not the validate and commit